server.go

```diff
@@ -838,29 +850,52 @@ func (s *Server) syncLocalOrderState() error {
 // determineBatchVersion determines the batch version that will be sent to the
 // auctioneer to signal compatibility with the different features the trader
 // client can support.
-func (s *Server) determineBatchVersion() order.BatchVersion {
+func (s *Server) determineBatchVersion() (order.BatchVersion, error) {
 	configVersion := s.cfg.DebugConfig.BatchVersion
 
 	// We set the default value of the config flag to -1, so we can
 	// differentiate between no value set and the first version (0).
 	if configVersion >= 0 {
-		return order.BatchVersion(configVersion)
+		return order.BatchVersion(configVersion), nil
 	}
 
-	isTaprootCompatible := false
-	verErr := lndclient.AssertVersionCompatible(
-		s.lndServices.Version, taprootVersion,
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), getInfoTimeout,
 	)
-	if verErr == nil {
-		isTaprootCompatible = true
+	defer cancel()
+
+	// We will try to decide what version to used based on the latest
+	// batch version + what features the node supports.
+	info, err := s.lndClient.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+	if err != nil {
+		return 0, err
 	}
 
-	// We can only auto-upgrade accounts to Taproot if the underlying node
-	// has the required capabilities.
-	if isTaprootCompatible {
-		return order.UpgradeAccountTaprootBatchVersion
+	// If the node supports ZeroConfChannels use that batch version.
+	_, zeroConfOpt := info.Features[uint32(lnwire.ZeroConfOptional)]
+	_, zeroConfReq := info.Features[uint32(lnwire.ZeroConfRequired)]
+	supportsZeroConf := zeroConfOpt || zeroConfReq
+
+	_, SCIDAliasOpt := info.Features[uint32(lnwire.ScidAliasOptional)]
+	_, SCIDAliasReq := info.Features[uint32(lnwire.ScidAliasRequired)]
+	supportsSCIDAlias := SCIDAliasOpt || SCIDAliasReq
+
+	if supportsZeroConf && supportsSCIDAlias {
+		return order.ZeroConfChannelsBatchVersion, nil
+	}
+
+	// If the node is taproot compatible use the
```

Comment:
We added a commit to set the min required version of `lnd` to be `v0.15.1-beta`. So we can return `order.UpgradeAccountTaprootBatchVersion` here directly and remove the `taprootVersion` variable at the top of the file.